<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>695</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0">
    <property name="leftMargin">
     <number>10</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <property name="bottomMargin">
     <number>10</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Generate</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,0,0,0,0,0,0,0">
          <item>
           <layout class="QHBoxLayout" name="majorRow">
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="majorLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Major root path:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="majorPath">
              <property name="toolTip">
               <string>The location of the root folder of major to run the generation.</string>
              </property>
              <property name="styleSheet">
               <string notr="true">padding: 3px</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pickMajorPath">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Open folder...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="sourceRow">
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="sourceLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Enforcer source file:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="sourceFile">
              <property name="toolTip">
               <string>The source file to mutate with major.</string>
              </property>
              <property name="styleSheet">
               <string notr="true">padding: 3px</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pickSourceFile">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Open file...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="mutationsOutputRow">
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="mutationsOutputLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Mutations output path:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="mutationsOutputPath">
              <property name="toolTip">
               <string>Where the generated major mutations will be stored.</string>
              </property>
              <property name="styleSheet">
               <string notr="true">padding: 3px</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pickMutationsOutputPath">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Open folder...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="mmlFileLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>MML settings file:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="mmlFile">
              <property name="styleSheet">
               <string notr="true">padding: 3px</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pickMmlFile">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Open file...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="checkMutationGroupFilter">
            <property name="text">
             <string>Use group mutation filter (this option disables MML settings file during generation if selected).</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="mutationsRow">
            <item>
             <widget class="QLabel" name="mutationsLabel">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Enabled mutations:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTreeWidget" name="mutationsTree">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="styleSheet">
               <string notr="true"/>
              </property>
              <property name="tabKeyNavigation">
               <bool>true</bool>
              </property>
              <property name="showDropIndicator" stdset="0">
               <bool>false</bool>
              </property>
              <property name="alternatingRowColors">
               <bool>true</bool>
              </property>
              <property name="selectionMode">
               <enum>QAbstractItemView::NoSelection</enum>
              </property>
              <property name="selectionBehavior">
               <enum>QAbstractItemView::SelectItems</enum>
              </property>
              <property name="indentation">
               <number>5</number>
              </property>
              <property name="sortingEnabled">
               <bool>false</bool>
              </property>
              <property name="headerHidden">
               <bool>true</bool>
              </property>
              <property name="columnCount">
               <number>1</number>
              </property>
              <attribute name="headerCascadingSectionResizes">
               <bool>false</bool>
              </attribute>
              <column>
               <property name="text">
                <string notr="true">1</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="generateButton">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>50</height>
             </size>
            </property>
            <property name="text">
             <string>Generate mutations</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Compile</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="mutationsInputRow">
            <property name="sizeConstraint">
             <enum>QLayout::SetMinimumSize</enum>
            </property>
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="mutationsInputLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Mutations input path:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="mutationsInputPath">
              <property name="styleSheet">
               <string notr="true">padding: 3px</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <property name="rightMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="pickMutationsInputPath">
                <property name="minimumSize">
                 <size>
                  <width>220</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="text">
                 <string>Select folder</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="copyMutationsPath">
                <property name="minimumSize">
                 <size>
                  <width>220</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="text">
                 <string>Copy from generation phase</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="enforcerRow">
            <property name="sizeConstraint">
             <enum>QLayout::SetMinimumSize</enum>
            </property>
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="enforcerLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Enforcer base project:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="enforcerPath">
              <property name="styleSheet">
               <string notr="true">padding: 3px</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pickEnforcerPath">
              <property name="minimumSize">
               <size>
                <width>220</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Select folder</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="apkOutputRow">
            <property name="sizeConstraint">
             <enum>QLayout::SetMinimumSize</enum>
            </property>
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="apkOutputLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>APK output path:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="apkOutputPath">
              <property name="styleSheet">
               <string notr="true">padding: 3px</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pickApkOutputPath">
              <property name="minimumSize">
               <size>
                <width>220</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Select folder</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="mutantsLogRow">
            <property name="sizeConstraint">
             <enum>QLayout::SetMinimumSize</enum>
            </property>
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="mutantsLogLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Mutants.log path:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="mutantsLogPath">
              <property name="styleSheet">
               <string notr="true">padding: 3px</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pickMutantsLogPath">
              <property name="minimumSize">
               <size>
                <width>220</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Select mutants.log file</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Exclude list</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QCheckBox" name="flagDisableLogs">
               <property name="text">
                <string>Disable compilation for XposedBridge.log -&gt; NOP replacements.</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="flagDisableHookMethod">
               <property name="text">
                <string>Disable compilation for findAndHookMethod -&gt; NOP replacements.</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="flagDisableStacktrace">
               <property name="text">
                <string>Disable compilation for e.printStackTrace -&gt; NOP replacements.</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <property name="topMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="label_2">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>150</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Drop mutations from source code line:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="dropFromLine">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>50</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">padding: 3px</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QLabel" name="estimatedCountLabel">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="compileButton">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>50</height>
             </size>
            </property>
            <property name="text">
             <string>Compile mutations to APKs</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Log</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Command log:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="logWindow">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="topMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="logClearButton">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>50</height>
             </size>
            </property>
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="styleSheet">
    <string notr="true">padding:0;margin:3px</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
